### PR TITLE
Various Overlay, Menu, and Away Mode bug fixes

### PR DESCRIPTION
**Pull request by ZappoMan**
- Fix several checkable menu items that would not properly trigger when checked via the Scripting API
- make the overlays toggle visible automatically when entering away mode
- enter away mode on the Vive if you remove the HMD

### CODE REVIEW

**Comment by ZappoMan** [comment]
Testing notes:

**Bug 1 - Away.js should automatically enter away mode when you take off the Vive**
https://app.asana.com/0/26225263936266/99845784533799
To test:
- Run this build with Vive
- Using [this version of away.js](https://rawgit.com/ZappoMan/hifi/toggleOverlay/examples/away.js)
- Remove Vive head set
- notice that you enter away mode

**Bug 2 - away mode should re-show overlays if they are hidden, because otherwise, you can't see the Away overlay note**
https://app.asana.com/0/26320649018758/99787385833930
To Test:
- Run this build
- Make sure you're running [this version of away.js](https://rawgit.com/ZappoMan/hifi/toggleOverlay/examples/away.js)
- Hide overlays using View > Overlays and uncheck them.
- Then do something to enter Away mode (press ESC, alt-tab from HMD, remove Vive HMD)
- Notice that the overlays should automatically show
- exit away mode - notice the overlays hide again.

**Bug 3 - some "toggle" menu items don't work from JS**
https://app.asana.com/0/26320649018758/99822748170839
To Test:
- Run this build
- Open Console
- enter: `Menu.setIsOptionChecked("Mute Microphone", false);` notice that the mic is unmuted.
- test with these menu items, they should be able to properly be toggled true/false and always work... Audio > Mute Microphone, View > Overlays, Settings > Developer Menus, Settings > Advanced Menus
- verify that all other checkable menu items also still work from script

**Comment by hifi-gustavo** [comment]
Build console output for commit: e341419090ea4305b123b0b7713e0d00fabc5711 available at: https://gist.github.com/371ab4af8d99c89cbf0b<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7326.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7326.dmg

File: libraries/ui/src/ui/Menu.cpp [hunk 1]
```diff
@@ -222,8 +222,11 @@ void Menu::setIsOptionChecked(const QString& menuOption, bool isChecked) {
         return;
     }
     QAction* menu = _actionHash.value(menuOption);
-    if (menu) {
-        menu->setChecked(isChecked);
+    if (menu && menu->isCheckable()) {
+        auto wasChecked = menu->isChecked();
+        if (wasChecked != isChecked) {
+            menu->trigger();
```
Comment (ZappoMan) [inline libraries/ui/src/ui/Menu.cpp]: note: this works for menu items that have slots connected to their triggered() signal to do things... which was the case for a handful of our checkable items

File: examples/away.js [hunk 2]
```diff
@@ -169,12 +179,21 @@ function goAway() {
     playAwayAnimation(); // animation is still seen by others
     showOverlay();
 
+    // remember the View > Overlays state...
+    wasOverlaysVisible = Menu.isOptionChecked("Overlays");
+    print("wasOverlaysVisible:" + wasOverlaysVisible);
```
Comment (jherico) [inline examples/away.js]: remove debug code?

Comment (ZappoMan) [inline examples/away.js]: gak!

**Comment by hifi-gustavo** [comment]
Build console output for commit: 608370615136d1b4d8ef57439632023b9bce53cd available at: https://gist.github.com/371ab4af8d99c89cbf0b<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7326.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7326.dmg

**Pull request merged**